python/lsst/ts/sal/make_idl_file.py

```diff
@@ -63,9 +63,22 @@ def copy_xml_files(self):
         ]
         sal_xml_paths = glob.glob(os.path.join(interfaces_dir, self.name, "*.xml"))
         if not sal_xml_paths:
-            raise RuntimeError(
-                f"Could not find any XML files for SAL component {self.name}"
+            print(
+                f"Could not find any XML files for SAL component {self.name}. "
+                "Component will only contain generics."
             )
+            dummy_xml_event_file = os.path.join(
+                self.sal_work_dir, f"{self.name}_Events.xml"
+            )
+            with open(dummy_xml_event_file, "w") as fp:
+                fp.write(
+                    """<?xml version="1.0" encoding="UTF-8"?>
+<?xml-stylesheet type="text/xsl" href="http://lsst-sal.tuc.noao.edu/schema/SALEventSet.xsl"?>
+<SALEventSet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://lsst-sal.tuc.noao.edu/schema/SALEventSet.xsd">
+</SALEventSet>
+"""
+                )
+
```

Comment:
Please see if this `if not sal_xml_paths:` is needed -- especially the parts that writes the dummy event file -- and if not, just delete it. I think this worked around a bug in ts_sal's generator but I'm hoping that bug has been fixed.

It probably makes sense to print  the message, though I'd reword it a bit to make sure it sounds perfectly normal, for example "Note: SAL component {self.name} only uses generic topics".

Reply:
Unfortunately, it is needed. salgenerator fails if the file is not there.. :(